src/main/java/org/elasticsearch/cluster/IncompatibleClusterStateVersionException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.ElasticsearchException;
+
+/**
+ * Thrown by {@link org.elasticsearch.cluster.Diffable#readDiffs(org.elasticsearch.common.io.stream.StreamInput, org.elasticsearch.cluster.node.DiscoveryNode)} method
+ */
+public class IncompatibleClusterStateVersionException extends ElasticsearchException {
+    public IncompatibleClusterStateVersionException(String msg) {
```

Comment:
Nit picky: I wonder if we should make this ctor get all the parameters and construct the message locally. Will be easier to read the code and use, imho.